File: bindings/go/binding_test.go
package tree_sitter_cfml_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_cfml "github.com/tree-sitter/tree-sitter-cfml/bindings/go"
)

func TestCanLoadCFMLGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_cfml.LanguageCFML())
	if language == nil {
		t.Errorf("Error loading CFML grammar")
	}
}

func TestCanLoadCFHTMLGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_cfml.LanguageCFHTML())
	if language == nil {
		t.Errorf("Error loading CFHTML grammar")
	}
}

func TestCanLoadCfscriptGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_cfml.LanguageCFSCRIPT())
	if language == nil {
		t.Errorf("Error loading CFSCRIPT grammar")
	}
}

File: bindings/node/binding_test.js
const assert = require('node:assert');
const { test } = require('node:test');

const Parser = require('tree-sitter');

test('can load CFML grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./cfml')));
});

test('can load CFHTML grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./cfhtml')));
});

test('can load CFSCRIPT grammar', () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require('./cfscript')));
});
